cobc/cobc.c

```diff
@@ -6533,16 +6481,103 @@ line_has_listing_statement (char *line, const enum cb_format source_format)
 	return 1;
 }
 
+static char *
+get_word (char *str)
+{
+	int i = 0;
+
+	if (!str) {
+		return NULL;
+	}
+	while (str[i] && isspace (str[i])) {
+		++i;
+	}
+	if (str[i]) {
+		return str + i;
+	}
+	return NULL;
+}
+
+static char *
+get_next_word (char *str)
+{
+	int i = 0;
+
+	if (!str) {
+		return NULL;
+	}
+	while (str[i] && !isspace (str[i])) {
+		++i;
+	}
+	while (str[i] && isspace (str[i])) {
+		++i;
+	}
+	if (str[i]) {
+		return str + i;
+	}
+	return NULL;
+}
+
+
+static COB_INLINE COB_A_INLINE int
+is_debug_line (char *line, int fixed, int acudebug)
+{
+	if (line == NULL || line[0] == 0) {
+		return 0;
+	}
+	return !cb_flag_debugging_line
+		&& ((fixed && line[cobc_get_indicator ()] == 'D')
+		    || (!fixed && (acudebug
+				   ? !strncasecmp (line, "\\D", 2)
+				   : !strncasecmp (line, "D ", 2))));
+}
+
+static COB_INLINE COB_A_INLINE int
+is_comment_line (char *line, int fixed)
+{
+	if (line == NULL || line[0] == 0) {
+		return 0;
+	} else {
+		const int indicator = cobc_get_indicator ();
+		return fixed
+			? line[indicator] == '*' || line[indicator] == '/'
+			: !strncmp (line, "*>", 2);
+	}
+}
+
+static void
+abort_if_too_many_continuation_lines (int pline_cnt, const char *filename, int line_num)
+{
+	if (pline_cnt >= CB_READ_AHEAD) {
+		cobc_err_msg (_("%s:%d: too many continuation lines"),
+				filename, line_num);
+		cobc_abort_terminate (0);
+	}
+}
+
+static void
+cleanup_copybook_reference (struct list_files *cur)
+{
+	if (cur->name) {
+		cobc_free ((void *)cur->name);
+	}
+	cobc_free (cur);
+}
+
 static void
 print_fixed_line (const int line_num, char pch, char *line)
 {
 	int		i;
-	int		len = strlen (line);
+	int		len;
 	const int	max_chars_on_line = cb_listing_wide ? 112 : 72;
 	const char	*format_str;
 	const int	indicator = cobc_get_indicator ();
 	const int	text_column = cobc_get_text_column ();
 
+	if (!line) {
```

Comment:
all callers have a guaranteed line, so that check should be dropped - please revert the change to `print_fixed_line` if that's correct

Reply:
Yes, effectively it's useless. I removed that.